src/anyp/Uri.cc

```diff
@@ -695,23 +648,18 @@ AnyP::Uri::absolute() const
 
         absolute_.append(getScheme().image());
         absolute_.append(":",1);
-        if (getScheme() != AnyP::PROTO_URN) {
-            absolute_.append("//", 2);
-            const bool allowUserInfo = getScheme() == AnyP::PROTO_FTP ||
-                                       getScheme() == AnyP::PROTO_UNKNOWN;
-
-            if (allowUserInfo && !userInfo().isEmpty()) {
-                static const CharacterSet uiChars = CharacterSet(UserInfoChars())
-                                                    .remove('%')
-                                                    .rename("userinfo-reserved");
-                absolute_.append(Encode(userInfo(), uiChars));
-                absolute_.append("@", 1);
-            }
-            absolute_.append(authority());
-        } else {
-            absolute_.append(host());
-            absolute_.append(":", 1);
+        absolute_.append("//", 2);
+        const bool allowUserInfo = getScheme() == AnyP::PROTO_FTP ||
+                                   getScheme() == AnyP::PROTO_UNKNOWN;
```

Comment:
This needs to exclude URI with image() of "urn:" (which is now part of `AnyP::PROTO_UNKNOWN`) or we open a security vulnerability for sensitive data exfiltration.

Reply:
Also, these changes are what render ICAP and helpers unable to meet the release notes claimed capability of performing Trivial-HTTP Resolver gateway.

Reply:
> This needs to exclude URI with image() of "urn:" (which is now part of `AnyP::PROTO_UNKNOWN`) or we open a security vulnerability for sensitive data exfiltration.

PR code treats all unknown (to Squid) URI schemes the same. This code had received unknown non-URN schemes prior to PR changes. Thus, the "we _open_ a vulnerability" assertion is false: Either that vulnerability existed before these changes, or these changes do not open it.



> Also, these changes are what render ICAP and helpers unable to meet the release notes claimed capability of performing Trivial-HTTP Resolver gateway.

That problem was flagged and addressed in [another change request](https://github.com/squid-cache/squid/pull/1930#discussion_r1826456197). If necessary, let's continue this discussion there.